tls/s2n_kem_preferences.c

```diff
@@ -15,31 +15,63 @@
 
 #include "tls/s2n_kem_preferences.h"
 
-const struct s2n_kem *pq_kems_r3_2021_05[1] = {
+const struct s2n_kem *pq_kems_r3_2021_05[] = {
     /* Round 3 Algorithms */
     &s2n_kyber_512_r3,
 };
 
-const struct s2n_kem_group *pq_kem_groups_r3[] = {
+const struct s2n_kem_group *pq_kem_groups_r3_2021_05[] = {
 #if EVP_APIS_SUPPORTED
     &s2n_x25519_kyber_512_r3,
 #endif
     &s2n_secp256r1_kyber_512_r3,
 };
 
+const struct s2n_kem_group *pq_kem_groups_r3_2023_06[] = {
+#if defined(S2N_LIBCRYPTO_SUPPORTS_KYBER)
+    &s2n_secp256r1_kyber_768_r3,
+    #if EVP_APIS_SUPPORTED
+    &s2n_x25519_kyber_768_r3,
+    #endif
+    &s2n_secp384r1_kyber_768_r3,
+    &s2n_secp521r1_kyber_1024_r3,
+#endif
+    &s2n_secp256r1_kyber_512_r3,
+#if EVP_APIS_SUPPORTED
+    &s2n_x25519_kyber_512_r3,
+#endif
+};
```

Comment:
I'll just keep telling myself that https://github.com/aws/s2n-tls/issues/4100 is coming 🥲

Reply:
it is. the implementation is underway in PR #4101.